crates/miden-lib/src/block/mod.rs

```diff
@@ -0,0 +1,115 @@
+use miden_core::Word;
+use miden_objects::ProposedBlockError;
+use miden_objects::block::{BlockBody, BlockHeader, BlockNumber, ProposedBlock};
+use miden_objects::transaction::PartialBlockchain;
+
+use crate::transaction::TransactionKernel;
+
+/// Builds a [`BlockHeader`] and [`BlockBody`] by computing the following from the state updates
+/// encapsulated by the provided [`ProposedBlock`]:
+/// - the account root;
+/// - the nullifier root;
+/// - the note root;
+/// - the transaction commitment; and
+/// - the chain commitment.
+///
+/// This functionality is handled here because the block header requires [`TransactionKernel`] for
+/// its various commitment fields.
+pub fn build_block(
+    proposed_block: ProposedBlock,
+) -> Result<(BlockHeader, BlockBody), ProposedBlockError> {
```

Comment:
In the next PR we'll probably add the `signature` parameter to this function (to include it in the block header).